fix(oas-utils): handles object in examples

antlio opened the pull request:
**Problem**
Currently, using the following in a document  `"example": ["Portfolio1", "Portfolio2"]` gets returned as an object in the request example.

**Solution**
this pr handles that use case and fixes #3782  in order to turn the object into an array to maintain its original type.

**Checklist**

I’ve gone through the following:

- [x] I’ve added an explanation _why_ this change is needed.
- [x] I’ve added a changeset (`pnpm changeset`).
- [x] I’ve added tests for the regression or new feature.

hanspagel commented on packages/oas-utils/src/spec-getters/getExampleFromSchema.test.ts:
```diff
@@ -1071,4 +1071,13 @@ describe('getExampleFromSchema', () => {
+    expect(
+      getExampleFromSchema({
+        type: 'array',
+        examples: [{ 0: 'example 1', 1: 'example 2' }],
```
That syntax seems strange and is not what’s in the PR description, pls enlighten me :D

antlio replied:
seems like using

```yaml
example:
  - example 1
  - example 2
```

is turned into a schema `examples` object, can sync about this if need!

antlio replied:
oops updated

amritk commented on packages/oas-utils/src/spec-getters/getExampleFromSchema.ts:
```diff
@@ -132,9 +132,12 @@ export const getExampleFromSchema = (
-    return cache(schema, schema.examples[0])
+    const firstExample = schema.examples[0]
+    return typeof firstExample === 'object'
+      ? cache(schema, Object.values(firstExample))
```
what if the example is supposed to be an object? Wouldn't we want the whole object?

antlio replied:
good call i guess it should remain, this was a first quick shot but i'll dig deeper regarding the casting

hanspagel reviewed (approved):
Actually, this didn’t work for strings, numbers and objects with the `examples` (the new way to provide multiple examples in OpenAPI 3.1).

I’ve added some more tests and fixed the code. ✌️

antlio commented:
@hanspagel i still get an object returned vs array

> Actually, this didn’t work for strings, numbers and objects with the `examples` (the new way to provide multiple examples in OpenAPI 3.1).
> 
> I’ve added some more tests and fixed the code. ✌️

I think i'm getting the object returned again instead of the array in the request example

hanspagel reviewed (changes requested):
I broke it

hanspagel reviewed (changes requested):
I think it’s working now. But I think it’s not fixing the initial bug report.

It looks like there’s somewith wrong in createRequestOperation. Before merging, we need to reproduce the issue in main and make sure it’s fixed here.

antlio commented:
closing this in favor of #4444
